docs/txp.md

```diff
@@ -0,0 +1,334 @@
+1. If `verifyAllIsKnown` is `True`, address' attributes must be known.
+2. If `verifyAllIsKnown` is `True`, address' type must be known.
+3. Address can't be a redeem address.
+
+### Consistency check
+
+There is only one consistency check: number of inputs in a transaction
+must be the same as number of witnesses for this transaction.
+
+### Transaction attributes check
+
+If `verifyAllIsKnown` is `True`, transaction's attributes must be known.
+
+### Inputs checks
+
+Other checks heavily depend on whether there is at least one input
+with unknown type. There are three cases:
+
+1. There is such input and `verifyAllIsKnown` is `True`. In this case
+   transaction is immediately considered invalid.
+2. There is such input and `verifyAllIsKnown` is `False`. It means
+   that there was a protocol update not known to our software. In this
+   case we can't know whether that input is valid and how much value
+   it has. So we perform only basic
+   checks: [outputs checks](#outputs-checks)
+   and [consistency check](#consistency-check). Note that we don't
+   perform
+   [transaction's attributes check](#transaction-attributes-check),
+   because `verifyAllIsKnown` is `False`.
+3. All inputs have known format (i. e. `TxInUtxo`). In this case
+   (which should be the most common one) we can check all inputs'
+   legitimacy, compare input and output sums, check fee, etc. All
+   inputs must exist in UTXO. In the following we describe the case
+   when all inputs have `TxInUtxo` type and we assume that for each
+   input we know corresponding unspent output (if we don't,
+   transaction is invalid).
+
+All inputs of a transaction must be different and each input must be
+properly certified by its witness.
+
+### Witness checks
+
+Witness checks consist of two parts: we need to check that witness
+corresponds to the address from the output we want to spend and we
+need to check that witness itself is correct.
+
+* If witness is `PkWitness` (which contains a public key and a
+  signature), we require that address is a `PubKey` address and its
+  spending data is `PubKey` spending data with the same key as in
+  the witness. Witness itself is checked by checking validity of the
+  signature stored in this witness. Note: signature inside witness is
+  given for `TxSigData` (which is basically the same as `Hash Tx`)
+  with `SignTx` tag.
+* If witness is `ScriptWitness` (which contains validator and redeemer
+  scripts), we require that address is a `Script` address and its
+  spending data is `Script` spending data with validator script from
+  the witness. Witness itself is valid if few conditions are met.
+  * Both redeemer and validator script have the same version.
+  * If `verifyAllIsKnown` is `True`, script version (which is same for
+    redeemer and validator) must be known. Currently it means that
+    version must be equal to 0, it's the only version known to our software.
+  * Plutus validation script build from redeemer and validator for
```

Comment:
I don't get this sentence, maybe you wanted to write
>Plutus validation script **built** from redeemer and validator

Reply:
Yes.